File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(count):
    the_sum = 0

    for i in range(count + 1):
        the_sum = the_sum + i

    return the_sum * the_sum

def sum_of_squares(count):
    the_sum = 0

    for i in range(count + 1):
        the_sum = the_sum + (i**2)

    return the_sum

def difference(count):
    sum_of_sq = sum_of_squares(count)

    sq_of_sum = square_of_sum(count)

    return sq_of_sum - sum_of_sq

File: c/difference-of-squares/src/difference_of_squares.c
#include "difference_of_squares.h"

int square_of_sum(int range)
{
        int result = 0;

        for (int i = 1; i <= range; i++) {
                result += i; 
        }

        return result * result; 
}

int sum_of_squares(int range)
{
        int result = 0; 

        for (int i = 1; i <= range; i++) {
                result += (i * i); 
        }

        return result; 
}

int difference_of_squares(int range)
{
        return square_of_sum(range) - sum_of_squares(range); 
}